oauth2client/client.py

```diff
@@ -1222,6 +1229,31 @@ def create_scoped(self, scopes):
         """
         return self
 
+    @classmethod
+    def from_json(cls, s):
+        # This import needs to be here to avoid a circular dependency.
+        from oauth2client.service_account import _ServiceAccountCredentials
+        data = json.loads(_from_bytes(s))
+
+        # We handle service_account._ServiceAccountCredentials since it is a
+        # possible return type of GoogleCredentials.get_application_default()
+        if (data['_module'] == 'oauth2client.service_account' and
+            data['_class'] == '_ServiceAccountCredentials'):
+            return _ServiceAccountCredentials.from_json(s)
```

Comment:
Re-surfacing the outdated comments here.

----

From me:

I think a more appropriate comment would be

```
# We handle service_account._ServiceAccountCredentials since it is a
# possible return type of GoogleCredentials.get_application_default()
```

----

From @nathanielmanistaatgoogle:

I think this is the sticking point for me - I'm a strong adherent to the "a superclass' having any knowledge of any of its subclasses is an abstraction violation" design principle. Are you also an adherent? If so, do you agree that there's a problem here? If there's a problem here, are we making it much worse with this change or only a little worse? If there's a problem here, is there a fix we can manage now in this change or should we put it off until later?